usaspending_api/disaster/v2/views/agency/loans.py

```diff
@@ -43,47 +45,110 @@ def route_agency_loans_backend(request, *args, **kwargs):
         self.filters.update({"award_type_codes": ["07", "08"]})
-        return self.perform_elasticsearch_search(loans=True)
 
-    def build_elasticsearch_result(self, info_buckets: List[dict]) -> List[dict]:
-        results = []
-        for bucket in info_buckets:
-            results.append(self._build_json_result(bucket))
-        return results
+        covid_faba_spending_by_toptier_agency = self._get_agency_covid_faba_spending()
+        json_result = self._build_json_result(covid_faba_spending_by_toptier_agency)
+        sorted_json_result = self._sort_json_result(json_result)
+
+        return Response(sorted_json_result)
+
+    def _get_agency_covid_faba_spending(self) -> QuerySet:
+        """
+        Query the covid_faba_spending table and return COVID-19 FABA spending grouped by toptier agencies
+        """
+
+        queryset = (
+            CovidFABASpending.objects.filter(spending_level="subtier_agency")
+            .filter(award_type__in=self.filters["award_type_codes"])
+            .filter(defc__in=self.filters["def_codes"])
+            .values("funding_toptier_agency_id", "funding_toptier_agency_code", "funding_toptier_agency_name")
+            .annotate(
+                award_count=Sum("award_count"),
+                obligation_sum=Sum("obligation_sum"),
+                outlay_sum=Sum("outlay_sum"),
+                face_value_of_loan=Sum("face_value_of_loan"),
+            )
+        )
 
-    def _build_json_result(self, bucket: dict):
-        return {
-            "id": int(bucket["key"]),
-            "code": bucket["dim_metadata"]["hits"]["hits"][0]["_source"]["funding_toptier_agency_code"],
-            "description": bucket["dim_metadata"]["hits"]["hits"][0]["_source"]["funding_toptier_agency_name"],
-            "children": [],
-            # the count of distinct awards contributing to the totals
-            "award_count": int(bucket["count_awards_by_dim"]["award_count"]["value"]),
-            **{
-                key: Decimal(bucket.get(f"sum_{val}", {"value": 0})["value"])
-                for key, val in self.nested_nonzero_fields.items()
-            },
-            "face_value_of_loan": bucket["count_awards_by_dim"]["sum_loan_value"]["value"],
+        if self.query:
+            queryset = queryset.filter(funding_toptier_agency_name__icontains=self.query)
+
+        return queryset
+
+    def _build_json_result(self, queryset: List[dict]) -> dict:
+        """Build the JSON response that will be returned for this endpoint.
+
+        Args:
+            queryset: Database query results.
+
+        Returns:
+            Formatted JSON response.
+        """
+
+        response = {}
+
+        results = [
+            {
+                "id": int(row["funding_toptier_agency_id"]),
+                "code": row["funding_toptier_agency_code"],
+                "description": row["funding_toptier_agency_name"],
+                "children": [],  # This type of spending will never have children
```

Comment:
Good inline comment. It does not tell the viewer what the code does, but tells us why it's an empty list.